scripts/var_props.py

```diff
@@ -808,6 +808,19 @@ class VarCompatObj:
                      _DOCTEST_RUNENV) #doctest: +ELLIPSIS
     <var_props.VarCompatObj object at 0x...>
 
+    # Test that a 2-D var with no vertical transform works
```

Comment:
this is a constant point of confusion for me - I'm mirroring the tests above and below that say that a 3D variable has a horizontal and vertical dimension and a 2D variable has only a horizontal dimension.

Reply:
Well ... it's super non-intuitive I suppose, but: In *most* modeling systems, a single horizontal dimension still represents a 2-dim distribution of grid columns (just stored in one long vector). This makes a lot of sense for CCPP, which is supposed to work with models that have regular and irregular grids. For example, in MPAS there's only one horizontal dimension, because the data is completely irregular. The convention/requirement therefore is that any model collapses the horizontal dimension(s) into one horizontal dimension for CCPP.

Unfortunately, there are other models like single column models where one could have individual columns for whatever horizontal distribution (none at all, same column, or meaningless, or actual columns picked from a 2-d distribution from another model); for example to test perturbation of physics for the same column.

For the vertical dimension, there's of course only one. So fields that have no horizontal dimension are definitely 1-d and not 2-d.

Reply:
Thanks Dom! That makes (vague) sense. I've updated the comments.